unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/WorldState.cs

```diff
@@ -77,6 +77,50 @@ public Vector3 ConvertSceneToUnityPosition(Vector3 pos, ParcelScene scene = null
             return solvedPosition;
         }
 
+        public Vector3 ConvertScenePositionToUnityPosition(ParcelScene scene = null)
+        {
+            if (scene == null)
+            {
+                WorldState worldState = Environment.i.worldState;
+                string sceneId = worldState.currentSceneId;
+
+
+                if (!string.IsNullOrEmpty(sceneId) && worldState.loadedScenes.ContainsKey(sceneId))
+                    scene = worldState.loadedScenes[worldState.currentSceneId];
+                else
+                    return Vector3.zero;
+            }
+
+            Vector3 scenePosition = Utils.GridToWorldPosition(scene.sceneData.basePosition.x, scene.sceneData.basePosition.y);
+            Vector3 worldPosition = DCLCharacterController.i.characterPosition.WorldToUnityPosition(scenePosition);
+
+            return worldPosition;
```

Comment:
Let's try to avoid copy/pasting code and use polymorphism properly.

All this block can be changed to `return ConvertScenePositionToUnityPosition(Vector3.zero, scene);`